services/server/src/handlers/user.ts

```diff
@@ -316,7 +316,11 @@ async function findUsers(req: Request, res: Response, next: NextFunction) {
 
   const query: IUser.FindUsersApiQuery = findUsersQuery.parse(req.query);
   const result = await users.find(query);
-  const response: IUser.FindUsersApiResponse = result;
+
+  const onlineUsers = await cache.onlineStatus.getAll();
+  result.list = result.list.map(user => ({...user, online: !!onlineUsers[user.id]}))
```

Comment:
It might not be a good idea to get all online users list to know if a subset of them are online or not. A better solution will be extend the `OnlineStatus` cache with method that accepts user ids and return if these users are online or not.